internal/olm/client/client.go

```diff
@@ -48,6 +48,22 @@ var ErrOLMNotInstalled = errors.New("no existing installation found")
 
 var Scheme = scheme.Scheme
 
+// custom error struct to capture deployment errors
+// while verifying CSV installs.
+type deploymentError struct {
+	name  string
+	issue string
+}
+type deploymentVerifyError []deploymentError
```

Comment:
I'm not a fan of call this `deploymentVerifyError` but I don't have a better suggestion right now.